ludwig/decoders/generic_decoders.py

```diff
@@ -37,6 +38,10 @@ def __init__(self, input_size: int = 1, num_classes: int = None, **kwargs):
     def forward(self, inputs, **kwargs):
         return inputs
 
+    @staticmethod
```

Comment:
Just brainstorming here: maybe `@register_encoder` and `@register_decoder` could take the schema class as an optional argument, and the registry could provide an accessor to get the schema for specified encoder/decoder type.

Or alternatively we could register the schema class `@register_encoder_schema`? 

To me, something like:
```
get_encoder_schema(feature_type, encoder)
```

is more readable than
```
get_encoder_cls(feature_type, encoder).get_schema_cls()
```

Reply:
This is a fair point. I would personally go for the first option in an effort to avoid more siloed registries. Will make the change now.

Reply:
After thinking a bit more, one thing to consider is that the method `get_schema_cls()` is used in input_features, output_features, combiners, encoders, and decoders. If I change it for the encoders, I would want to change it for the rest of the modules to remain consistent. This won't be really tricky but before I do this, just want to make sure that this is the way we want to do this. Any thoughts @dantreiman? If anyone else has thoughts on this as well, I'd love to hear them.

Reply:
Thanks for the comprehensive review. I appreciate the diligence:)